code/app/src/main/java/com/cmput301w21t06/crowdfly/Database/UserController.java

```diff
@@ -52,23 +57,62 @@ public void onEvent(@NonNull QuerySnapshot response, @Nullable FirebaseFirestore
      */
     public static void getUsers(CrowdFlyListeners.OnDoneGetIdsListener onDoneGetIdsListener) {
         ArrayList<String> ids = new ArrayList<String>();
-        for (User user : users.values()){
-            ids.add(user.getUserID());
+        for (String id : converter.keySet()){
+            ids.add(id);
         }
+        Log.e("sleep",String.valueOf(ids));
         onDoneGetIdsListener.onDoneGetIds(ids);
     }
 
     /**
-     * This finds a particular user by their id
-     * @param uid
-     * The user id we are searching for
+     * This finds a particular user by their display id
+     * @param did
+     * The user's display id we are searching for
      * @param onDoneGetUserListener
      * The class that implements the onDone listener
      */
-    public static void getUserProfile(String uid, CrowdFlyListeners.OnDoneGetUserListener onDoneGetUserListener){
-        onDoneGetUserListener.onDoneGetUser(users.get(uid));
+    public static void getUserProfile(String did, CrowdFlyListeners.OnDoneGetUserListener onDoneGetUserListener){
+        Log.e("sleep",String.valueOf(did));
+        onDoneGetUserListener.onDoneGetUser(users.get(converter.get(did)));
+    }
 
+    public static String reverseConvert(String uid){
+        for (String k : converter.keySet()){
+            if (converter.get(k).matches(uid)){
+                return k;
+            }
+        }
+        return "ERROR";
+    }
 
+    /**
+     * This faciliates the creation of a new user in the database
+     * @param user
+     * This is the user object that is partially created locally
+     */
+    public static void addUserProfile(@NonNull User user){
+        GodController.getDb().runTransaction(new Transaction.Function<Void>() {
+            @Override
+            public Void apply(@NonNull Transaction transaction) throws FirebaseFirestoreException {
+                DocumentReference doc = GodController.getDb().document(CrowdFlyFirestorePaths.counter());
+                Long place = transaction.get(doc).getLong("counter");
+                Long did = place;
+                user.setDisplayID(String.valueOf(did));
+                transaction.update(doc,"counter",place + 1);
```

Comment:
Not super important but this is an easier way of updating a counter.
https://firebase.google.com/docs/reference/js/firebase.firestore.FieldValue#increment